modules/base/src/datavisualizer/imageinformationvisualizer.cpp

```diff
@@ -0,0 +1,103 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/base/datavisualizer/imageinformationvisualizer.h>
+
+#include <inviwo/core/common/inviwoapplication.h>
+#include <modules/base/processors/imagesource.h>
+#include <modules/base/processors/imageinformation.h>
+#include <inviwo/core/processors/processorutils.h>
+#include <inviwo/core/ports/imageport.h>
+
+#include <inviwo/core/io/datareaderfactory.h>
+
+namespace inviwo {
+
+using GP = util::GridPos;
+
+ImageInformationVisualizer::ImageInformationVisualizer(InviwoApplication* app)
+    : DataVisualizer{}, app_(app) {}
+
+std::string ImageInformationVisualizer::getName() const { return "Image Information"; }
+
+Document ImageInformationVisualizer::getDescription() const {
+    Document doc;
+    auto b = doc.append("html").append("body");
+    b.append("", "Construct an image information processor");
+    return doc;
+}
+
+std::vector<FileExtension> ImageInformationVisualizer::getSupportedFileExtensions() const {
+    auto rf = app_->getDataReaderFactory();
+    auto exts = rf->getExtensionsForType<Layer>();
```

Comment:
Not image?

Reply:
generally a file on disk would correspond to a Layer not an Image